File: easy/age_distribution.hs
import System.Environment (getArgs)

age  :: Int -> String
age i | i < 0 || i > 100 = "This program is for humans"
      | i < 3            = "Still in Mama's arms"
      | i < 5            = "Preschool Maniac"
      | i < 12           = "Elementary school"
      | i < 15           = "Middle school"
      | i < 19           = "High school"
      | i < 23           = "College"
      | i < 66           = "Working for the man"
      | otherwise        = "The Golden Years"

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (age . read) $ lines input

File: easy/age_distribution.lua
local category = {"This program is for humans", "Still in Mama's arms", "Preschool Maniac", "Elementary school", "Middle school", "High school", "College", "Working for the man", "The Golden Years"}
local age = {0, 3, 5, 12, 15, 19, 23, 66, 101}

for line in io.lines(arg[1]) do
    local m = 0
    while m < 9 and tonumber(line) >= age[m+1] do
        m = m + 1
    end
    print(category[(m%9)+1])
end
